### PR TITLE
Active channel cleanup

### PR DESCRIPTION
**Pull request by seanyesmunt**
Did some refactoring so we can more easily use the `<ChannelSelector />` component

- This will make channel signing (required for new comment apis) much simpler
- Now has one active channel app wide, which isn't tied to comments
- Anonymous actions will happen by enabling the new `incognito` value which can be used to pause search/watch history when that is added back.

### CODE REVIEW

File: ui/component/comment/view.jsx [hunk 1]
```diff
@@ -121,11 +110,6 @@ function Comment(props: Props) {
   } catch (e) {}
 
   useEffect(() => {
-    // If author was extracted from the URI, then it must be valid.
-    if (authorUri && author && !isResolvingUri && shouldFetch) {
-      resolveUri(authorUri);
```
Comment (seanyesmunt) [inline ui/component/comment/view.jsx:126]: This wasn't being used so I took it out.

File: ui/component/publishName/view.jsx [hunk 2]
```diff
@@ -1,51 +1,41 @@
 // @flow
```
Comment (seanyesmunt) [inline ui/component/publishName/view.jsx:1]: This was in the same component as PublishBid. I separated them to move the name field up to the top of the publish form

**Review by jessopb — COMMENTED** [review]
Seems like the publish confirmation grabbing the channel?
It works on odysee.

File: ui/component/app/index.js [hunk 3]
```diff
@@ -45,6 +58,8 @@ const perform = dispatch => ({
   getWalletSyncPref: () => dispatch(doGetWalletSyncPreference()),
   syncLoop: noInterval => dispatch(doSyncLoop(noInterval)),
   setReferrer: (referrer, doClaim) => dispatch(doUserSetReferrer(referrer, doClaim)),
+  setActiveChannelIfNotSet: () => dispatch(doSetActiveChannel()),
+  setIncognito: () => dispatch(doSetIncognito()),
```
[the remark this answers is not in the document]
Comment (seanyesmunt) [reply ui/component/app/index.js:62]: It won't be labled in the UI as that. I thought it might be more useful as a variable name than `anonymous`.

**Comment by seanyesmunt** [comment]
Fixed the issue with publish confirmation not picking it up. Merging once the build passes

**Pull request merged**